raiden_contracts/data/source/services/ServiceRegistry.sol

```diff
@@ -3,49 +3,167 @@ pragma solidity 0.5.4;
 import "raiden/Token.sol";
 import "raiden/Utils.sol";
 
+contract Deposit {
```

Comment:
Why is this it's own contract? Couldn't you also model this as a struct inside the `ServiceRegistry` contract?

Reply:
Yes, everything could be in `ServiceRegistry` and that would be gas-efficient.

The current choice follows security-in-depth.  Even if `ServiceRegistry` is compromized, the deposits made earlier would not be affected.

Is this conceptual security worth more than the additional gas?  I don't know for sure, but the deposits are rarer and bigger than usual Raiden transfers.

Reply:
That argument makes sense to me, but I don't have much experience. The question was aimed at understanding the trade-off better!